src/libstd/collections/hash/map.rs

```diff
@@ -353,37 +352,50 @@ fn search_hashed<K, V, M, F>(table: M,
     // undefined behavior when Bucket::new gets the raw bucket in this
     // case, immediately return the appropriate search result.
     if table.capacity() == 0 {
-        return TableRef(table);
+        return InternalEntry::TableIsEmpty;
```

Comment:
As far as I can tell, no one can ever handle `TableIsEmpty` without panicking. Perhaps `search_hashed` should just require and assert that `table.capacity() > 0` and `InternalEntry` should be simplified to not include `TableisEmpty`.